spec/client_spec.rb

```diff
@@ -308,11 +347,14 @@ def redis.with
           # This time we do not pass it through to Redis, in order to simulate a passing
           # call to LOAD SCRIPT followed by another NOSCRIPT error. Imagine someone
           # repeatedly calling SCRIPT FLUSH on our Redis instance.
-          expect(@manipulated_instance).to receive(:load_scripts)
+          allow(@manipulated_instance).to receive(:load_scripts)
 
           expect {
             lock_manager.lock(resource_key, ttl)
-          }.to raise_error(/NOSCRIPT/)
+          }.to raise_error do |e|
+            expect(e.errors[0]).to be_a(RedisClient::CommandError)
```

Comment:
let's also add assertion that there's just one error:

```ruby
expect(e.errors.count).to eq 1
```

ditto to other specs working on `e.errors`